sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/context/AzureTokenCredentialAutoConfiguration.java

```diff
@@ -222,11 +222,10 @@ static AzureServiceClientBuilderFactoryPostProcessor builderFactoryBeanPostProce
         return new AzureServiceClientBuilderFactoryPostProcessor();
     }
 
-    @SuppressWarnings("removal")
     @Bean(name = DEFAULT_CREDENTIAL_TASK_EXECUTOR_BEAN_NAME)
     @ConditionalOnMissingBean(name = DEFAULT_CREDENTIAL_TASK_EXECUTOR_BEAN_NAME)
     ThreadPoolTaskExecutor credentialTaskExecutor() {
-        return new TaskExecutorBuilder()
+        return new ThreadPoolTaskExecutorBuilder()
```

Comment:
Since we are targeting to support as more as possible Spring Boot versions, and the `ThreadPoolTaskExecutorBuilder` is introduced since 3.2.0, so we are going to change it to this:

```java
ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
scheduler.setPoolSize();

return scheduler;
```